backend/src/main/java/com/github/redstylzz/backend/service/PaymentService.java

```diff
@@ -30,44 +34,60 @@ private boolean paymentExists(String paymentID) {
         return paymentRepo.existsByPaymentID(paymentID);
     }
 
+    private void calculatePaymentSum(String userID, String categoryID) {
+        List<Payment> payments = paymentRepo.getAllByUserIDAndCategoryID(userID, categoryID);
+        BigDecimal sum = payments.stream().map(Payment::getAmount).reduce(BigDecimal.ZERO, BigDecimal::add);
+        categoryService.setCategorySum(userID, categoryID, sum);
+    }
+
+    private List<PaymentDTO> getPaymentAsDTO(String userID, String categoryID) {
+        return paymentRepo
+                .getAllByUserIDAndCategoryID(userID, categoryID)
+                .stream()
+                .map(Payment::convertPaymentToDTO)
+                .toList();
+    }
 
-    public List<Payment> getPayments(String userID, String categoryID) {
-        return paymentRepo.getAllByUserIDAndCategoryID(userID, categoryID);
+    public List<PaymentDTO> getPayments(String userID, String categoryID) {
+        return getPaymentAsDTO(userID, categoryID);
```

Comment:
not quite sure why you split this method into two parts 
if you would to split it, I would rather have one method for your data repository interaction (to retrieve all values as list) and then convert them into an DTO when handing them over to the requesting service. This way if your DTO structure changes only one method is affected by that (the one handing over the DTO) and your repository interaction remains unchanged